### ISSUE TITLE
SW: Decide on a default web browser

### ISSUE THREAD

**Issue by parazyd** [maemo-leste/bugtracker#22]

**Comment by MerlijnWajer** [comment]
Maybe surf from suckless can provide a simple base

**Comment by NotKit** [comment]
There is http://doc.qt.io/QtWebBrowser/ (hosted at http://code.qt.io/cgit/qt-apps/qtwebbrowser.git/) which could be used as base for browser with mobile UI. Requires at least 1 GB RAM.

**Comment by pkillnine** [comment]
The Nokia N900 only has 256MB of RAM

**Comment by MerlijnWajer** [comment]
Mind you that Fremantle also has a lot of swap. We have zram, and we can also add some swap. I think it is a reality that nearly all the browsers are somewhat slow. I wouldn't expect it to actually use that much when you start/use it, so might be worth trying it out.

**Comment by dderby** [comment]
Indeed.  It's pretty much a given that modern web browsing will be slow on any device with low memory like the N900.  I suggest bundling 2 web browsers on low memory devices.  A very lightweight one like Dillo which does not support JavaScript and one a little more beefy with a modern engine like QtWebBrowser, Midori or Surf, even though they will be slow.

I actually quite like Midori.  It has a web app mode which hides the UI (and uses less memory) and makes it easy to package web apps which are launched from an icon on the desktop or menu.  I currently do this under Fremantle with Uber for example.

Zram will help.  I found that it was not enabled by default.  This should fix that:
https://github.com/maemo-leste/image-builder/pull/1

**Comment by ollieparanoid** [comment]
[**NetSurf**](http://netsurf-browser.org/) is a lightweight browser that runs without problems on the N900.

_The only problem is, that I wouldn't trust it to be safe against exploits from the web. So I'd be happy to see that with a wrapper that does proper [privilege separation](https://en.wikipedia.org/wiki/Privilege_separation) (i.e. the NetSurf process can't access anything from the system or network without asking a broker that double checks everything for permission, and you'd have one NetSurf process per tab). But that would be a project on its own._

**Comment by dderby** [comment]
This may be useful to help us decide on a browser:
https://www.ssllabs.com/ssltest/viewMyClient.html

**Comment by MerlijnWajer** [comment]
I've been using surf2, and even on the n900 it's quite fast/smooth. The scrollbar could be better, we might want to add scrolling by gesture, and customise the dmenu url editing. But for basic usage/integration, it might not be such a bad choice for now.

**Comment by MerlijnWajer** [comment]
Shall we just try to pick one that works decenty? Bonuspoints if it works with our vkb. Then we can at least just provide one.

**Comment by MerlijnWajer** [comment]
netsurf-gtk and midori both seem ok. They integrate with our gtk theme and even the virtual keyboard seems to work. I think I have a slight preference for netsurf-gtk, but yeah. Shall we just pick one?

**Comment by MerlijnWajer** [comment]
Moving to alpha, since I think it should be simple enough to just install netsurf-gtk or midori as a starting point.

**Comment by parazyd** [comment]
I don't think netsurf does justice to Leste. People would just find it
weird.

**Comment by MerlijnWajer** [comment]
@parazyd  this is one of the last tickets before the alpha milestone is done, what do you recommend we make the default?
